stunnel/stunnel.ml

```diff
@@ -180,14 +188,19 @@ let config_file config extended_diagnosis host port =
        ; ( match config with
          | None ->
              []
-         | Some {sni; cert_bundle_path} ->
+         | Some {sni; verify; cert_bundle_path} ->
              [
                ""
              ; "# use SNI to request a specific cert. CAfile contains"
              ; "# public certs of all hosts in the pool and must contain"
              ; "# the cert of the server we connect to"
              ; (match sni with None -> "" | Some s -> sprintf "sni = %s" s)
-             ; "verifyPeer=yes"
+             ; ( match verify with
+               | VerifyPeer ->
+                   "verifyPeer=yes"
+               | CheckHost ->
+                   sprintf "checkHost=%s" host
```

Comment:
Reading stunnel's manual doesn't seem like using checkHost means that `verifyPeer = yes` is used as well. I think having both options defined in the case is good to have.